recipes/args/all/conanfile.py

```diff
@@ -0,0 +1,108 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class ArgsConan(ConanFile):
+    name = "args"
+    description = "Small open-source library for program argument parser, inspired by Python's `argparse`, " \
+                  "depending only on the standard library, with C++17 as minimum requirement."
+    homepage = "https://github.com/mbits-libs/args"
+    license = "MIT"
+    topics = ("command-line", "commandline", "commandline-interface",
+              "program-arguments", "argparse", "argparser", "argument-parsing")
+
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    generators = "cmake"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        compiler = str(self.settings.compiler)
+        if compiler in self._working_compilers:
+            working_compiler = self._working_compilers[compiler]
+            current_compiler = ArgsConan.Version(
+                str(self.settings.compiler.version))
+            if working_compiler > current_compiler:
+                msg = f"args library requires {compiler} version to be at least " \
+                      f"{working_compiler}; this configuration uses version {current_compiler}"
+                raise ConanInvalidConfiguration(msg)
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("{}-{}".format(self.name, self.version),
+                  self._source_subfolder)
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+    def _configure_cmake(self):
+        gen = CMake(self)
+        gen.definitions["LIBARGS_TESTING"] = "OFF"
+        gen.definitions["LIBARGS_INSTALL"] = "ON"
+        gen.definitions["LIBARGS_SHARED"] = "ON" if self.options.shared else "OFF"
+        gen.configure(source_folder=self._source_subfolder)
+        return gen
```

Comment:
cache `cmake` to avoid reconfiguration between build and package.

```suggestion
        if self._cmake is not None:
            return self._cmake
        self._cmake = CMake(self)
        self._cmake._cmakedefinitions["LIBARGS_TESTING"] = "OFF"
        self._cmake.definitions["LIBARGS_INSTALL"] = "ON"
        self._cmake.definitions["LIBARGS_SHARED"] = "ON" if self.options.shared else "OFF"
        self._cmake.configure(source_folder=self._source_subfolder)
        return self._cmake
```

Reply:
Question: is `_cmakedefinitions` in the proposed snippet a hasty editing artifact? I'll assume it is and stick to `definitions`...

Reply:
> self._cmake._cmakedefinitions["LIBARGS_TESTING"] = False

Was most certainly a type! Nice catch

Reply:
Yes that was a typo! :blush: 
I fixed it.

Reply:
Done